picard/ui/options/maintenance.py

```diff
@@ -0,0 +1,166 @@
+
+        # Setting options from all option pages and loaded plugins (including plugins currently disabled).
+        key_options = set(config.setting.as_dict())
+
+        # Combine all page and plugin settings with required options not appearing in option pages.
+        current_options = set([
+            # Include options that are required but are not entered directly from the options pages.
+            'file_renaming_scripts',
+            'selected_file_naming_script_id',
+            'log_verbosity',
+            # Items missed if TagsCompatibilityWaveOptionsPage does not register.
+            'remove_wave_riff_info',
+            'wave_riff_info_encoding',
+            'write_wave_riff_info',
+        ]).union(key_options)
+
+        # All setting options included in the INI file.
+        config.beginGroup("setting")
+        file_options = set(config.childKeys())
+        config.endGroup()
+
+        orphan_options = file_options.difference(current_options)
+
+        self.ui.description.setText(
+            _(self.INSTRUCTIONS)
+            + _("\n\nThe configuration file currently contains {0} option settings, {1} which are unused.").format(
+                len(file_options),
+                len(orphan_options)
+            )
+        )
+        self.ui.enable_cleanup.setChecked(False)
+        self.ui.tableWidget.clearContents()
+        self.ui.tableWidget.setRowCount(len(orphan_options))
+        for row, option_name in enumerate(sorted(orphan_options)):
+            tableitem = QtWidgets.QTableWidgetItem(option_name)
+            tableitem.setData(QtCore.Qt.UserRole, option_name)
+            tableitem.setFlags(tableitem.flags() | QtCore.Qt.ItemIsUserCheckable)
+            tableitem.setCheckState(False)
+            self.ui.tableWidget.setItem(row, 0, tableitem)
+            tableitem = QtWidgets.QTextEdit()
+            text = self.make_setting_value_text(option_name)
+            tableitem.setPlainText(text)
+            tableitem.setReadOnly(True)
+            # Adjust row height to reasonably accommodate values with more than one line, with a minimum
+            # height of 25 pixels.  Multi-line values will be expanded to display up to 5 lines, assuming
+            # a standard line height of 18 pixels.
+            row_height = max(25, 18 * min(5, text.count("\n") + 1))
+            self.ui.tableWidget.setRowHeight(row, row_height)
+            self.ui.tableWidget.setCellWidget(row, 1, tableitem)
+        self.ui.tableWidget.resizeColumnsToContents()
+        self.ui.select_all.setCheckState(False)
+        self.enable_cleanup_changed()
+
+    def save(self):
+        if not self.ui.enable_cleanup.checkState() == QtCore.Qt.Checked:
+            return
+        to_remove = set()
+        for idx in range(self.ui.tableWidget.rowCount()):
+            item = self.ui.tableWidget.item(idx, 0)
+            if item.checkState() == QtCore.Qt.Checked:
+                to_remove.add(item.data(QtCore.Qt.UserRole))
```

Comment:
Perhaps we could break down this method a bit and use a generator, something like:

```python
    def column_items(self, column):
        for idx in range(self.ui.tableWidget.rowCount()):
            yield self.ui.tableWidget.item(idx, column)

    def selected_options(self):
        for item in self.column_items(0):
            if item.checkState() == QtCore.Qt.Checked:
                yield item.data(QtCore.Qt.UserRole)

    def select_all_changed(self):
        state = self.ui.select_all.checkState()
        for item in self.column_items(0):
            item.setCheckState(state)

    def save(self):
        if not self.ui.enable_cleanup.checkState() == QtCore.Qt.Checked:
            return
        to_remove = set(self.selected_options())
```

It doesn't change stuff much, but results in smaller methods, which is usually better.